Service/DataService.php

```diff
@@ -2010,4 +2023,101 @@ public function createAndMapIndex(Environment $environment): void
+                                        ]
+                                    ],
+                                    [
+                                        'terms' => [
+                                            $contentType->getBusinessIdField() => $items
+                                        ]
+                                    ],
+                                ]
+                            ]
+                        ]
+
+                    ],
+                    'size' => 100,
+                    "scroll" => "1m"
+                ]);
+
+                while (count($result['hits']['hits'] ?? []) > 0) {
+                    foreach ($result['hits']['hits'] as $hits) {
+                        $key = sprintf('%s:%s', $contentType->getName(), $hits['_id']);
+                        $ouuids[$hits['_source'][$contentType->getBusinessIdField()]] = $key;
+                        $this->cacheOuuids[$contentTypesCommaList][$contentType->getBusinessIdField()] = $key;
+                    }
+                    $result = $this->client->scroll([
+                        'scroll_id' => $result['_scroll_id'],
+                        'scroll' =>  "1m",
+                    ]);
+                }
+            }
+        }
+        return array_values($ouuids);
+    }
+
+    public function getDataLink(string $contentTypesCommaList, string $businessId): ?string
+    {
+        return $this->getDataLinks($contentTypesCommaList, [$businessId])[0] ?? $businessId;
+    }
+
+
+
+    public function hitFromBusinessIdToDataLink(ContentType $contentType, string $ouuid, array $rawData) : Document
+    {
+        $revision = $this->getEmptyRevision($contentType, null);
+        $revision->setRawData($rawData);
+        $revision->setOuuid($ouuid);
+        $revisionType = $this->formFactory->create(RevisionType::class, $revision, ['migration' => true, 'raw_data' => $revision->getRawData(), 'with_warning' => false]);
+        $result = $this->walkRecursive($revisionType->get('data'), $rawData, function (string $name, $data, DataFieldType $dataFieldType, DataField $dataField) {
+            if ($data !== null && (!is_array($data) || count($data) > 0)) {
+                if ($dataFieldType->isVirtual()) {
+                    return $data;
+                }
+
+                if (!$dataFieldType instanceof DataLinkFieldType) {
+                    return [$name => $data];
+                }
+
+                $typesList = $dataField->getFieldType()->getDisplayOption('type');
+                if ($typesList == null) {
+                    return [$name => $data];
+                }
+
+                if (is_string($data)) {
+                    return [$name => $this->getDataLink($typesList, $data)];
+                }
+                return [$name => $this->getDataLinks($typesList, $data)];
+            }
+            return [];
+        });
+        unset($revisionType);
```

Comment:
Is the garbage collection of php not sufficient?

Reply:
It was like that I'll do some tests a1b274c09f67c45f305002653d5d75168b8c56ae